gadget/layout.go

```diff
@@ -189,6 +193,69 @@ func LayoutVolumePartially(volume *Volume, constraints LayoutConstraints) (*Part
 	return vol, nil
 }
 
+func resolveOne(gadgetRootDir, kernelRootDir string, kernelInfo *kernel.Info, pathOrRef string) (string, error) {
+	// content may refer to "$kernel:<name>/<content>"
+	if strings.HasPrefix(pathOrRef, "$kernel:") {
+		kernelRef := pathOrRef[len("$kernel:"):]
```

Comment:
Perhaps `regexp.Regexp.ReplaceAllFunc` and expand `$[a-z]+` this way. I think we have similar problems in other places (interface attributes, particularly in relation to content). Would be good to do something similar here.

Reply:
I updated this now to use a regexp.